engine/worker_task_queue.go

```diff
@@ -164,7 +165,24 @@ func (w *workerTaskQueueImpl) pollAndDispatchAndPrepareNext() {
 
 func (w *workerTaskQueueImpl) commitCompletedPages(ctx context.Context) error {
 	if len(w.completedPages) > 0 {
-		// TODO optimize by combining all pages into single query (as long as it won't exceed certain limit)
+		// merge pages, e.g.,
+		// [1, 2], [3, 4], [7, 8] -> [1, 4], [7, 8]
```

Comment:
Nit: maybe put this code in a separate method

Reply:
also it's better to have a "unit test" for this

Reply:
Yeah this is better.